src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolverInterface.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Responsible for the value of an argument based on its metadata.
+ *
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+interface ArgumentValueResolverInterface
+{
+    /**
+     * Should return true if this resolver can resolve the value for the given ArgumentMetadata.
+     *
+     * @param Request          $request
+     * @param ArgumentMetadata $argument
+     *
+     * @return bool
+     */
+    public function supports(Request $request, ArgumentMetadata $argument);
+
+    /**
+     * Yield the possible value(s).
+     *
+     * An implementation must yield at least one value.
```

Comment:
If it is handled properly (by throwing or whatever) this comment line might just be useless, isn't it ?